prompt_poet/template.py

```diff
@@ -43,6 +46,14 @@ def __init__(
                 self._template_name,
             ) = self._parse_template_path(template_path, from_examples=from_examples)
         self._package_name = package_name
+        if template_loader:
+            self._template_loader = template_loader
+        else:
+            # To be backward compatible.
+            if package_name:
+                self._template_loader = LocalPackageTemplateLoader(package_name, template_path)
+            else:
+                self._template_loader = LocalFSTemplateLoader(template_path)
```

Comment:
can you read this again and make sure it is correct? 

specifically
```
        if template_path:
            (
                self._template_dir,
                self._template_name,
            ) = self._parse_template_path(template_path, from_examples=from_examples)
```

but then if no template_loader you do this 
```
            if package_name:
                self._template_loader = LocalPackageTemplateLoader(package_name, template_path)
            else:
                self._template_loader = LocalFSTemplateLoader(template_path)
```

are self._template_dir going and self._template_name going to be used at all downstream or are you just going to use self._template_loader?

Reply:
right, I intend to just use the self._template_loader downstream. self._template_dir and self._template_name are kept for backward compatibility.

Reply:
The example path were not handled right, fixed that.